.travis.yml

```diff
@@ -69,37 +60,15 @@ install:
-- if [ "$RUN_E2E_TESTS_TOPICS_AND_SKILLS_DASHBOARD" == 'true' ]; then travis_retry python -m scripts.run_e2e_tests --suite="topicsAndSkillsDashboard" --prod_env; fi
-- if [ "$RUN_E2E_TESTS_TOPIC_AND_STORY_EDITOR" == 'true' ]; then travis_retry python -m scripts.run_e2e_tests --suite="topicAndStoryEditor" --prod_env; fi
-
-
-cache:
-  # Cache Oppia's dependencies.
-  directories:
-    - ../oppia_tools/
-    - node_modules/
-    - third_party/
-    - $HOME/.cache/pip
-    - $HOME/.cache/TravisChrome/
-    - backend_prod_files/templates/
-    - build/templates/
-    - backend_prod_files/extensions/
-    - build/extensions/
+- if [ "$RUN_E2E_TESTS_ADDITIONAL_EDITOR_PLAYER_FEATURES" == 'true' ]; then python -m scripts.run_e2e_tests --suite="additionalEditorFeatures" --prod_env; python -m scripts.run_e2e_tests --skip-build --skip-install --suite="additionalPlayerFeatures" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_COMMUNITY_DASHBOARD" == 'true' ]; then python -m scripts.run_e2e_tests --suite="communityDashboard" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_CORE_EDITOR_AND_PLAYER_FEATURES" == 'true' ]; then python -m scripts.run_e2e_tests --suite="coreEditorAndPlayerFeatures" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_CREATOR_DASHBOARD" == 'true' ]; then python -m scripts.run_e2e_tests --suite="creatorDashboard" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_EXPLORATION_FEEDBACK_HISTORY_TAB" == 'true' ]; then python -m scripts.run_e2e_tests --suite="explorationFeedbackTab" --prod_env; python -m scripts.run_e2e_tests --skip-build --skip-install --suite="explorationHistoryTab" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_EXPLORATION_STATISTICS_TRANSLATION_TAB" == 'true' ]; then python -m scripts.run_e2e_tests --suite="explorationStatisticsTab" --prod_env; python -m scripts.run_e2e_tests --skip-build --skip-install --suite="explorationTranslationTab" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_EXTENSIONS" == 'true' ]; then python -m scripts.run_e2e_tests --suite="extensions" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_LEARNER" == 'true' ]; then python -m scripts.run_e2e_tests --suite="learner" --prod_env; python -m scripts.run_e2e_tests --skip-build --skip-install --suite="learnerDashboard" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_SKILL_EDITOR" == 'true' ]; then python -m scripts.run_e2e_tests --suite="skillEditor" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_TOPIC" == 'true' ]; then python -m scripts.run_e2e_tests --suite="topicAndStoryEditor" --prod_env; python -m scripts.run_e2e_tests --skip-build --skip-install --suite="topicsAndSkillsDashboard" --prod_env; fi
+- if [ "$RUN_E2E_ACCESSIBILITY_EMBEDDING" == 'true' ]; then python -m scripts.run_e2e_tests --suite="accessibility" --prod_env; python -m scripts.run_e2e_tests --skip-build --skip-install --suite="embedding" --prod_env; fi
 
-before_cache:
-  # Delete python bytecode to prevent cache rebuild.
-  - find third_party -name "*.pyc" -print -delete
-  - find backend_prod_files/extensions/ -name "*.pyc" -print -delete
```

Comment:
I removed caching since it doesn't serve any purpose since we're rebuilding from scratch every time. This should hopefully remove some flakes too.

Reply:
Do we download the entire oppia_tools and node_modules from scratch? I don't think so, are you sure? (As I think it takes a lot of time to download all the dependencies.)

Reply:
Actually, we already do that, the caching mechanism doesn't really do anything for our case since we call the install and build script every time. Plus, the caching mechanism doesn't make much sense because the build files are different for every PR (it's following from one PR to another).

Reply:
but the oppia_tools and third_party libs are mostly the same, right? i.e, we won't have to re-download everything if we cache?

Reply:
![image](https://user-images.githubusercontent.com/16653571/84850478-80653500-b075-11ea-8eb1-abfe1945e284.png)
I tried checking the build logs and I see this i.e, the script check for the tools and it already exists in the build so we don't have to download it!

Reply:
Good catch, I'll add those back.

Reply:
done!